WordPress/Classes/Utility/Editor/GutenbergSettings.swift

```diff
@@ -89,12 +101,25 @@ class GutenbergSettings {
         let blog = post.blog
 
         if post.isContentEmpty() {
-            return blog.editor.mobile == .gutenberg && blog.editor.web == .gutenberg
+            return shouldUseGutenbergForNewPosts(on: blog)
         } else {
             // It's an existing post
             return post.containsGutenbergBlocks()
         }
     }
+
+    private func shouldUseGutenbergForNewPosts(on blog: Blog) -> Bool {
+        guard let userSelectedEditor = blog.editor.mobile else {
+            return getDefaultEditor(for: blog) == .gutenberg
+        }
+
+        return userSelectedEditor == .gutenberg
+    }
+
+    func getDefaultEditor(for blog: Blog) -> MobileEditor {
+        // Default to gutenberg on WPCom/Jetpack sites
```

Comment:
I'm not sure I know the reason for this, but it'd be good to expand a bit on the comment for future reference